src/pipeline.cc

```diff
@@ -223,9 +226,11 @@ void Pipeline::UpdateFramebufferSizes() {
     return;
 
   for (auto& attachment : color_attachments_) {
-    attachment.buffer->SetWidth(fb_width_);
-    attachment.buffer->SetHeight(fb_height_);
-    attachment.buffer->SetElementCount(size);
+    auto mip0_width = fb_width_ << attachment.base_mip_level;
+    auto mip0_height = fb_height_ << attachment.base_mip_level;
+    attachment.buffer->SetWidth(mip0_width);
+    attachment.buffer->SetHeight(mip0_height);
+    attachment.buffer->SetElementCount(mip0_width * mip0_height);
```

Comment:
The frame buffer size does not change, but the first mip level of the attached image is larger than the framebuffer in case the base mip level is larger than zero. Should I still add a comment about that?

Reply:
Yes, the buffer is larger than the frame buffer. Currently the EXPECT command would check the first mip level (mip0_width x mip0_height in this case), as that's always at the beginning of the buffer. Would we want to add an optional MIP_LEVEL parameter for EXPECT so we could check each MIP level? Sounds like that would require getting the mip level buffer offset from Vulkan driver.

Reply:
Maybe file a bug for this and we can look at it if we need it in the future? As long as the comparisons do what's expected at this point it should be fine.